Update README.md (#23)
r0nsha/chili: README.md

Chili is a general-purpose, compiled programming language, focused on productivity, expressiveness and joy of programming™.

> At this stage, Chili brings a robust type system, bringing the syntax and fun of modern high-level programming language, with the performance characteristics of low-level language.

Reviewer: Maybe something like this could be better?
```suggestion
At this stage, Chili brings the syntax and fun of a modern, high-level programming language, a robust type system and the performance characteristics of a low-level language.

```

> **🌶 The language is still going through its early iterations.**  
> To build the compiler from source, you will need to:
>
> - [Set up](https://github.com/llvm/llvm-project/releases/tag/llvmorg-12.0.1) LLVM 12 on your machine.You should build from source under windows.
> - [Install](https://www.rust-lang.org/tools/install) the rust compiler

As there's no freestanding binary yet, we still need to build the compiler from source. To build the compiler, run:

```
cargo build
```

> **Note:** Although the compiler itself supports both Windows and Linux(Mac OS support is on the way), the standard library is very young, and doesn't support Windows yet. This will be implemented in the near future. Have a look at the [Tasks](#Tasks) section.

Reviewer: ```suggestion
**Note:** Although the compiler itself supports both Windows and Linux (Mac OS support is also planned), the standard library is very young, and doesn't support Windows yet. This will be implemented in the near future. Have a look at the [Tasks](#Tasks) section.

```

> **🌶 The language is still going through its early iterations.**  
> To build the compiler from source, you will need to:
>

> > - [Set up](https://github.com/llvm/llvm-project/releases/tag/llvmorg-12.0.1) LLVM 12 on your machine.You should build from source under windows.

Reviewer: Is this true? I think you can use Visual studio build tools for this

> > - [Set up](https://github.com/llvm/llvm-project/releases/tag/llvmorg-12.0.1) LLVM 12 on your machine.You should build from source under windows.

Reviewer: ```suggestion
> - [Set up](https://github.com/llvm/llvm-project/releases/tag/llvmorg-12.0.1) LLVM 12 on your machine. Windows users need to build from source.

```

- [ ] Runtime type introspection
- [ ] Printing and formatting
- [ ] Parametric polymorphism - supporting both types and constant values
- [ ] Associated functions / Methods
- [ ] Memory management model - undecided, but here are some options, from most likely to least likely:
  - [ ] Regions
  - [ ] Ownership + ARC
  - [ ] Ownership
  - [ ] Garbage collection
- [ ] Enums & Pattern matching
- [ ] Traits / Typeclasses
- [ ] Closures
- [ ] Named function arguments
- [ ] Code testing
- [ ] Error handling
- [ ] Multithreading

### Standard library
- [ ] OS Abstractions for Windows
  - [ ] Filesystem API
  - [ ] Networking and HTTP
  - [ ] Date/Time
- [ ] OS Abstractions for Linux
  - [ ] Filesystem API
  - [ ] Networking and HTTP
  - [ ] Date/Time
- [ ] OS Abstractions for MacOS
  - [ ] Filesystem API
  - [ ] Networking and HTTP
  - [ ] Date/Time
- [ ] Formatting/Printing

> - [ ] Basic data structure and algorithm

Reviewer: ```suggestion
- [ ] Basic data structures
  - [ ] Box
  - [ ] List/Vec/Seq (names are hard...)
  - [ ] String
  - [ ] HashMap
  - [ ] HashSet
```

- [ ] Runtime type introspection
- [ ] Printing and formatting
- [ ] Parametric polymorphism - supporting both types and constant values
- [ ] Associated functions / Methods
- [ ] Memory management model (The design is still in progress)
  - [ ] Ownership (Move & Copy semantics)
  - [ ] Safe references ("Borrowing")
- [ ] Enums & Pattern matching
  - [ ] Option & Result types
  - [ ] Try ? operator
- [ ] Traits / Typeclasses
- [ ] Closures
- [ ] Named function arguments
- [ ] Built-in code testing

### Standard library
- [ ] OS Abstractions for Windows
  - [ ] Filesystem API
  - [ ] Networking and HTTP
  - [ ] Date/Time
- [ ] OS Abstractions for Linux
  - [ ] Filesystem API
  - [ ] Networking and HTTP
  - [ ] Date/Time
- [ ] OS Abstractions for MacOS
  - [ ] Filesystem API
  - [ ] Networking and HTTP
  - [ ] Date/Time
- [ ] Formatting/Printing
- [ ] Basic data structures
  - [ ] Box

>   - [ ] List/Vec/Seq (It's hard to list them all)

Reviewer: ```suggestion
  - [ ] List/Vec/Seq (Haven't settled on a name yet)

```

- [x] Default function arguments
- [x] Panic function
- [x] Varargs

> - [x] Standard library support for Linux

Reviewer: The standard library isn't really a thing right now. It's in very very early stages (it doesn't even have a List/Vec or even a String struct) (Both will be added soon though).

I'd remove this

Reviewer: We could add a `Standard library` section with things that we're planning on implementing or have already implemented. For example:
```
## Standard library

- [ ] OS Abstractions
- [ ] Formatting/Printing
- [ ] Filesystem API
- [ ] Networking and HTTP
- [ ] Date/Time
etc...
```
